internal/server/handle_callback.go

```diff
@@ -0,0 +1,61 @@
+package server
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/jmartin82/mmock/pkg/mock"
+)
+
+// HandleCallback makes a callback required after a request
+func HandleCallback(cb mock.Callback) (*http.Response, error) {
+	if d := cb.Delay.Duration; d > 0 {
+		log.Printf("Delaying callback by: %s\n", d)
+		time.Sleep(d)
+	}
+
+	url := cb.Url
+	log.Printf("Making callback to %s\n", url)
+	req, err := http.NewRequest(cb.Method, url, bytes.NewBufferString(cb.Body))
+	if err != nil {
+		log.Printf("Error creating callback request: %s\n", err)
+		return nil, err
+	}
+	// add headers
+	for h, vs := range cb.Headers {
+		for _, v := range vs {
+			req.Header.Set(h, v)
+		}
+	}
+
+	client := &http.Client{Timeout: time.Second * 10}
```

Comment:
Would be nice have a constant instead of the number:

`const TIMEOUT_SECONDS = 10`

Or even better an optional configuration

Reply:
Good idea. I've added `timeout: Delay` to the callback structure, with a default of 10s where it's not set.

Reply:
This is great!